### PR TITLE
feat: payment details

### PR DESCRIPTION
**Pull request by Noums26**

### CODE REVIEW

File: src/operations/payment-methods/PaymentMethodForm.tsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  Switch,
+} from "@mui/material";
+import {PaymentElement, useElements, useStripe} from "@stripe/react-stripe-js";
+import {useCreate, useNotify} from "react-admin";
+import {PaymentMethodsAction} from "@jcloudify-api/typescript-client";
+
+export const PaymentMethodForm: React.FC<{onSuccess: () => void}> = ({
+  onSuccess,
+}) => {
+  const [isSubmiting, setIsSubmiting] = useState(false);
```
Comment (YumeT023) [inline src/operations/payment-methods/PaymentMethodForm.tsx:16]: already exposed in the 2nd element of the useCreate array
```typescript
[create, { is... }] = useCreate()
```

File: src/operations/payment-methods/PaymentMethodForm.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  const notify = useNotify();
+  const stripe = useStripe();
+  const elements = useElements();
+  const [create] = useCreate("paymentMethods");
+  const submitPaymentInfo = async (e: any) => {
```
Comment (YumeT023) [inline src/operations/payment-methods/PaymentMethodForm.tsx:21]: any ?

File: src/operations/payment-methods/PaymentMethodForm.tsx [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+  FormControl,
+  FormControlLabel,
+  Stack,
+  Switch,
+} from "@mui/material";
+import {PaymentElement, useElements, useStripe} from "@stripe/react-stripe-js";
+import {useCreate, useNotify} from "react-admin";
+import {PaymentMethodsAction} from "@jcloudify-api/typescript-client";
+
+export const PaymentMethodForm: React.FC<{onSuccess: () => void}> = ({
```
Comment (YumeT023) [inline src/operations/payment-methods/PaymentMethodForm.tsx:13]: use React admin `Create/CreateBase` directly

File: src/operations/payment-details/PaymentDetails.tsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+export const PaymentDetailsSummary: React.FC = () => {
+  const customerId = authProvider.getCachedWhoami()?.user?.stripe_id!;
+  const navigate = useNavigate();
+
+  return (
+    <Stack direction="column" spacing={1}>
+      <Stack direction="row" spacing={1} justifyContent="space-between">
+        <Typography variant="h6">Payment details</Typography>
+        <IconButtonWithTooltip
```
Comment (YumeT023) [inline src/operations/payment-details/PaymentDetails.tsx:26]: doesnt IconButtonWithTooltip expose a 'to' prop for navigation ? leave it as is if it doesnt

File: src/operations/payment-methods/BankCardShow.tsx [hunk 5]
```diff
@@ -0,0 +1,39 @@
+import {Box} from "@mui/material";
+import {CardNumberElement, Elements} from "@stripe/react-stripe-js";
+import {loadStripe} from "@stripe/stripe-js";
+import {colors} from "@/themes";
+
+export const BankCardShow: React.FC<{last4: string}> = ({last4}) => {
+  const stripePk = process.env.REACT_APP_STRIPE_PK;
```
Comment (YumeT023) [inline src/operations/payment-methods/BankCardShow.tsx:7]: check config/env.ts 

do not use process.env.thing in-code, put them in config/env.ts first

File: src/operations/payment-methods/BankCardShow.tsx [hunk 6]
```diff
@@ -0,0 +1,39 @@
+import {Box} from "@mui/material";
+import {CardNumberElement, Elements} from "@stripe/react-stripe-js";
+import {loadStripe} from "@stripe/stripe-js";
+import {colors} from "@/themes";
+
+export const BankCardShow: React.FC<{last4: string}> = ({last4}) => {
```
Comment (YumeT023) [inline src/operations/payment-methods/BankCardShow.tsx:6]: last4 -> last4Digit

File: src/operations/payment-methods/PaymentMethods.tsx [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+} from "@mui/icons-material";
+import {
+  PaymentMethod,
+  PaymentMethodsAction,
+} from "@jcloudify-api/typescript-client";
+import {authProvider} from "@/providers";
+import {colors} from "@/themes";
+import {BankCardShow} from "./BankCardShow";
+import {Elements} from "@stripe/react-stripe-js";
+import {PaymentMethodForm} from "./PaymentMethodForm";
```
Comment (YumeT023) [inline src/operations/payment-methods/PaymentMethods.tsx:26]: always use "@/stuff" location for imports

File: src/operations/payment-details/PaymentDetails.tsx [hunk 8]
```diff
@@ -0,0 +1,38 @@
+import {IconButtonWithTooltip} from "react-admin";
```
Comment (YumeT023) [inline src/operations/payment-details/PaymentDetails.tsx:1]: why is there payment-details, payment-methods, payments folder ?

delete all the folders about payments, only leave payments there, then put everything about payments in there
you are always allowed to create multiple providers, but not multiple operations module about a feature

**Review by YumeT023 — CHANGES_REQUESTED** [review]

File: src/operations/payment-methods/PaymentMethodForm.tsx (hunk 1 above)
Comment (Noums26) [reply src/operations/payment-methods/PaymentMethodForm.tsx:16]: Yes but is... remains unchanged during and after the operation.

File: src/operations/payment-methods/PaymentMethodForm.tsx (hunk 3 above)
Comment (Noums26) [reply src/operations/payment-methods/PaymentMethodForm.tsx:13]: The stripe element must be submitted first to obtain the ID.

**Pull request closed without merging**